parlai/core/worlds.py

```diff
@@ -456,32 +456,34 @@ def __next__(self):
         if self.epoch_done():
             raise StopIteration()
 
-    def batch_observe(self, index, batch):
-        for w in self.worlds:
+    def batch_observe(self, index, batch_actions):
+        batch_observations = []
+        for i, w in enumerate(self.worlds):
             agents = w.get_agents()
-            agents[index].observe(validate(batch[index]))
-        return batch
+            observation = agents[index].observe(validate(batch_actions[i]))
+            batch_observations.append(observation)
```

Comment:
Yes. Should we do that in this change, or when changing all the agents' returns?